html-aam/index.html

```diff
@@ -15435,6 +15435,19 @@ <h4 id="att-usemap">`usemap`</h4>
               <th>Comments</th>
               <td>Refer to <a href="#el-img">`img`</a> element.</td>
             </tr>
+            <tr tabindex="-1" id="att-validity">
+              <th>`validity [IDL]`</th>
+              <td class="elements">
+                HTML elements;
+                <a data-cite="html/#the-input-element:dom-cva-validity">`input`</a>
+              </td>
+              <td class="aria"><a class="core-mapping" href="#ariaInvalidTrue">`aria-invalid`</a> ="true"</td>
+              <td class="ia2">Use WAI-ARIA mapping</td>
+              <td class="uia">Use WAI-ARIA mapping</td>
+              <td class="atk">Use WAI-ARIA mapping</td>
+              <td class="ax">Use WAI-ARIA mapping</td>
+              <td class="comments">If the element has the `validity [IDL]` set and the `aria-invalid` attribute set, User Agents MUST expose only the`validity [IDL]` state.</td>
```

Comment:
I think el.ariaInvalid would stilll need to be exposed… in part because I don't know how to remove it with a HTMLInputElement  ILD override, so we could leave it and expose as null. @rahimabdi what do you think?

Reply:
It's not entirely clear to me what the requirement here is.

It's also not clear to me why tests couldn't be written for this.

Is the idea that if the form element is considered valid per HTML's requirements, `aria-invalid` is to be ignored? I think that should happen on the AT layer and attribute reflection shouldn't really be involved in that. (Thus if you set the `aria-invalid` attribute, the `ariaInvalid` getter will still reflect that, even if it's valid.)

Reply:
This comment is not about whether aria-invalid is reflected in javascript -- this is the html-aam, which means it is about how the user agent exposes that information to the accessibility API. The comment is about what to expose in the accessibility API when the IDL validity and aria-invalid conflict.

Reply:
Also we can't write tests for that yet -- but I'm working on it :)

Reply:
I suspected as much, but I continue to think this is very confusing. We shouldn't refer to public API as public API can be overwritten in JavaScript.

Instead you probably want to have some kind of algorithm:

> Is element _element_ to be considered valid by AT:
> 1. If _element_ is a form-associated element:
>    1. return true if it satisfies its constraints (https://html.spec.whatwg.org/#concept-fv-valid); otherwise false.
> 2. If it has an aria-invalid attribute, then return false.
> 3. Return true.

Reply:
I think I see your point about the algorithm, that is clear to me, except for the language `to be considered valid by AT` - this spec isn't about what ATs do, it is about what browsers expose over the accessibility APIs. So we have to use the language `User agents expose`, like in the original note.

Also I'm curious for my own understanding what you mean by "refer to public API"? What do you mean by public API?

Reply:
Anything defined in IDL. We shouldn't be calling that directly.